### ISSUE TITLE
Syntax Highlighters doesn't work properly with Gutenberg

### ISSUE THREAD

**Issue by Warlib1975** [WordPress/gutenberg#13522]
**Describe the bug**
I tried several Syntax Highlighters as block in Gutenberg and all of them convert special symbols like ", <, & to &qout; &lt; &amp and etc. They work the first time, then after post te record Gutenberg plugin told, that something is wrong and recommend to convert to HTML or fix the block and damage the source code completely. 
   
![image](https://user-images.githubusercontent.com/11045291/51783413-635f5780-214a-11e9-84e5-31714e90a76f.png)

I checked the syntax highlighter plugins checked with >5.0 version of WordPress (only two of such): 
https://wordpress.org/plugins/enlighter/
https://ru.wordpress.org/plugins/syntaxhighlighter/

And I have the problem, that Gutenberg decided to move such post with "broken" content to trash without asking me. :-( 

**To Reproduce**
Just place any С++ code with ", &, < after to any Syntax Highliter block and after saving it will be broken.

**Screenshots**
![image](https://user-images.githubusercontent.com/11045291/51783418-6ce8bf80-214a-11e9-9cab-e63f1e494290.png)

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser Chrome
 - Version 71.0.3578.98 (Official Build) (64-bit)

**Comment by swissspidy** [comment]
AFAIK https://wordpress.org/plugins/syntaxhighlighter/ works just fine with Gutenberg, so it sounds like the plugins you've been testing aren't 100% compatible with Gutenberg.

**Comment by youknowriad** [comment]
Looks like there's a path forward here. Thanks.

**Comment by Warlib1975** [comment]
> AFAIK https://wordpress.org/plugins/syntaxhighlighter/ works just fine with Gutenberg, so it sounds like the plugins you've been testing aren't 100% compatible with Gutenberg.

This plugin doesn't work properly. Here is chat with the author: https://github.com/Viper007Bond/syntaxhighlighter/issues/98#issuecomment-457936767

**Comment by youknowriad** [comment]
Got it so for the core/code block, this is a duplicate of #13218

**Comment by btdevdesign** [comment]
Still broken for me as well. Tested in the Classic Block and the Paragraph Block. 

The only way it works is in the 'syntaxhighlighter/code' Block and that cannot be combined with any paragraphs or images to float beside, etc.

Paragraph blocks -1st is using the SH shortcode, and the 2nd is just plain text (in editor):
<img width="599" alt="Screen Shot 2020-03-25 at 1 50 07 PM" src="https://user-images.githubusercontent.com/26233384/77584160-9496fd00-6e9f-11ea-9bbc-c7897b8d43e1.png">

Ends up like this. Shortcode breaks, plain text is fine (front end):
<img width="804" alt="Screen Shot 2020-03-25 at 1 48 23 PM" src="https://user-images.githubusercontent.com/26233384/77584168-99f44780-6e9f-11ea-9b93-8420ffb47562.png">

I noticed that if I turn the Syntaxhighlighter plugin off it DOES NOT change the code back. The text is being REWRITTEN for some reason.
